File: Aula04/Exemplo2/app/src/main/res/drawable/MainActivity.java
package com.example.aula04exemplo2;

import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import androidx.appcompat.app.AppCompatActivity;
import android.view.View;
import android.content.Intent;

public class MainActivity extends AppCompatActivity {
    private EditText editNome;
    private Button btnEnviar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        editNome = findViewById(R.id.editNome); // onde digitamos nosso nome
        btnEnviar = findViewById(R.id.btnEnviar); // botão

        btnEnviar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String nomeDigitado = editNome.getText().toString();

                Intent irParaSegundaTela = new Intent(MainActivity.this, SegundaActivity.class);
                irParaSegundaTela.putExtra("nomeUsuario", nomeDigitado);
                startActivity(irParaSegundaTela);
            }
        });
    }
}

File: aula4_exemplo2/app/src/main/java/com/example/aula4_exemplo2/MainActivity.kt
package com.example.aula4_exemplo2

import android.R
import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import androidx.appcompat.app.AppCompatActivity


class MainActivity : AppCompatActivity() {
    private var editNome: EditText? = null
    private var btnEnviar: Button? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        editNome = findViewById<EditText?>(R.id.editNome) // onde digitamos nosso nome
        btnEnviar = findViewById<Button?>(R.id.btnEnviar) // botão

        btnEnviar!!.setOnClickListener(object : View.OnClickListener {
            override fun onClick(v: View?) {
                val nomeDigitado = editNome!!.getText().toString()

                val irParaSegundaTela = Intent(this@MainActivity, SegundaActivity::class.java)
                irParaSegundaTela.putExtra("nomeUsuario", nomeDigitado)
                startActivity(irParaSegundaTela)
            }
        })
    }
}